### DISCUSSION TITLE
Preferred method of communication?

### DISCUSSION THREAD

**Discussion by kirby-koo** [wolfi-dev/os#5186] [General]
Coming off of the September community call, we discussed what the best mode of communication for the Wolfi community. Suggestions were Slack, Discord, Mastodon & GitHub Discussions. Please share your thoughts!

**Comment by amouat** [comment]
I don't really follow the mastodon thought - how would that work? Is there chat functionality & is it searchable?

With Discord I like the idea of using it to run the Wolfi community call as well.

**Comment by blaisep** [comment]
@kirby-koo , IMHO, It depends on what kind of communication we want to have:

### Instant messaging (ie. realtime text/video chat)
- Slack
- Discord
- Matrix

### Microblog (short form stream of consciousness)
- Mastodon
- Twitter/X (Yuck, count me out)

### Discussions (long form forums)
- Discourse (super customizable and now includes embedded chat)
- Github discussion
- Reddit (... meh ...)
- StackOverflow

---
FWIW, my bias is against walled gardens.
Discourse and Matrix are open source and both have active development communities.
Matrix is a little heavy handed with the security, but that aligns with this audience...

**Reply by polarathene** [reply]
**TL;DR:** Matrix of Discord 👍 

---

Discord has channels for chatting, which can also support temporal threads to minimize noise in a channel when there's multiple discussions interleaving.

Additionally Discord has a feature to do long form discussion isolated from a real-time chat channel, where you create a thread/post and wait for someone to reply over a longer duration.

Both types are indexed for search AFAIK, so it should meet needs pretty well.

---

Matrix is probably good too, but you'll likely find Discord has a wider audience presently, so if you want to focus on a particular platform Discord is a better bet? You could consider bridging the two 🤷‍♂️ 

---

I don't find much interest in Mastodon / Twitter platforms for support/community engagement personally. I think they're fine for the community themselves to interact socially, but not it's rarely ever been a good resource for me to get assistance or find information from that the other platforms don't do better.

Reddit isn't bad for engaging with the community on some topics and getting some assistance, and I often get some pages there when using a search engine. I don't really see it as an ideal platform for a project to manage a community through IMO. StackOverflow fits into the same category.

Github Discussions as a developer is quite convenient due to the majority of us already having a Github account and potentially interacting with a project via issues/PR already. I maintain a project where support requests are deferred to discussions and it's been good for some other projects to manage announcements of new releases/betas and get feedback from community. It's usually not my preferred platform but it works alright (_at least on desktop, never tried it on my phone like I do with Discord_).

Discourse is generally nice when I come across that as a user. Main pain point with it is that I rarely want to signup for another service, which is probably common enough for users that it adds some friction. Unlike Discord, where one account can join many community discord servers.

Slack may be convenient for some if they're already using it. I rarely do, last use was last year to engage with Docker maintainers and Slack was not as pleasant as Discord to use, I recall some UX issues that surprised me.

**Comment by amouat** [comment]
Closing this discussion for the moment.
